nighthawk/source/common/utility.cc

```diff
@@ -29,26 +29,73 @@ uint32_t determineCpuCoresWithAffinity() {
 
 } // namespace PlatformUtils
 
-Uri::Uri(const std::string& uri) : scheme_("http") {
+size_t Uri::findPortSeparatorInAuthority(absl::string_view authority) {
+  size_t colon_index = std::string::npos;
+  bool in_ipv6_address = false;
+  bool last_seen_dot = true;
+
+  // TODO(oschaaf): We need to revisit this and improve. This performs some
+  // limited hostname validation.
+  for (size_t i = 0; i < authority.size(); i++) {
+    char c = authority[i];
+    if (i == 0 && c == '[') {
+      in_ipv6_address = true;
```

Comment:
I'm probably super dense here, so apologies if we've covered this before, but what's the reason we can't use `Utility::parseInternetAddressAndPort` again?